src/osd/OSD.h

```diff
@@ -197,40 +197,11 @@ class OSDService : public Scrub::ScrubSchedListener {
   std::map<epoch_t, unsigned> map_reservations;
 
   /// gets ref to next_osdmap and registers the epoch as reserved
-  OSDMapRef get_nextmap_reserved() {
-    std::lock_guard l(pre_publish_lock);
-    epoch_t e = next_osdmap->get_epoch();
-    std::map<epoch_t, unsigned>::iterator i =
-      map_reservations.insert(std::make_pair(e, 0)).first;
-    i->second++;
-    return next_osdmap;
-  }
+  OSDMapRef get_nextmap_reserved();
   /// releases reservation on map
-  void release_map(OSDMapRef osdmap) {
```

Comment:
Does the first commit contain any fix or it's just about shuffling the code + extra debugs?   If the later, saying this explicitly in the commit description would be helpful.

Reply:
Done